Bring back the EventListenerOrEventListenerObject and EventListener as an extendable interface

orta opened the pull request:
[Angular](https://typescript.visualstudio.com/TypeScript/_build/results?buildId=105317&view=logs&j=71179031-6417-5a2f-3d87-af6fce2011e4&t=5ac81f15-0a74-5a9b-b155-06ef07cd84f6) relies on `EventListenerOrEventListenerObject` which makes changing this name basically a no-go I'm afraid

saschanaz commented:
I know this is basically what the code was, but I'd prefer adding them in addedTypes.json and removing the original callback from removedTypes.json rather than complicating the build function. What do you think?

orta commented:
I can give it a shot 👍🏻

saschanaz commented:
I'd really like to convert those JSONs as JSON5/YAML or any other commentable formats but out of scope here 😬

orta commented:
I added the ability to write comments in the JSON too, and basically apologized a bunch in the code.

Rough changes:
 - Needed to add call signatures to an interface (removed in https://github.com/saschanaz/types-web/pull/110/commits/e2f82399a6ab0a43f70762d393a0219c071013a5 and replaced with a commented hack) 
 -  Some commented why but "mother wouldn't be proud" code in the renaming process

saschanaz commented on inputfiles/addedTypes.json:
```diff
@@ -1154,6 +1189,13 @@
+            {
+                "name": "EventListenerOrEventListenerObject",
+                "type": [
+                    { "type": "_EventListener" },
```
Hey, there is `override-type` to workaround this 😄

orta replied:
Good point! Rebased this on main and fix that up

saschanaz commented on inputfiles/addedTypes.json:
```diff
@@ -557,6 +557,41 @@
+                        // This is a hack to add a call signature, but I think it's reasonable
+                        // as it means we don't have to add a call signatures section to the
+                        // emitter for this one case.
+                        "callable": {
```
Clever 😄

orta commented:
Makes sense to me 👍🏻

saschanaz commented on inputfiles/knownTypes.json:
```diff
@@ -30,6 +31,13 @@
+        "RsaKeyAlgorithm",
+        "RsaKeyGenParams",
+        "RsaOaepParams",
+        "RsaPssParams",
```
I think this one is initially sorted by `.sort()` and thus case-sensitive, but probably okay.

saschanaz commented on src/build/emitter.ts:
```diff
@@ -404,6 +405,7 @@ export function emitWebIdl(
     if (baseTypeConversionMap.has(objDomType)) {
       return baseTypeConversionMap.get(objDomType)!;
     }
+
```
```suggestion
```

saschanaz commented:
Thanks! 😍
